### PR TITLE
Add Docstrings

### PR DESCRIPTION
**Pull request by josephsdavid**
- [x] Gaussian general docstrings
- [x] Gaussian example
- [x] Multinomial general docstrings 
- [x] Multinomial example (Simple bag of words???)

### CODE REVIEW

File: src/MLJNaiveBayesInterface.jl [hunk 1]
```diff
@@ -141,4 +141,125 @@ MMI.is_pure_julia(::Type{<:MultinomialNBClassifier}) = true
 MMI.input_scitype(::Type{<:MultinomialNBClassifier}) = Table(Count)
 MMI.target_scitype(::Type{<:MultinomialNBClassifier}) = AbstractVector{<:Finite}
 
+"""
+$(MMI.doc_header(GaussianNBClassifier))
+
+
+`GaussianNBClassifier`: Gaussian Naive Bayes classifier. Assumes variables have a
+multivariate normal distribution. Good for real-valued data.
```
Comment (tlienart) [inline src/MLJNaiveBayesInterface.jl]: nit: the second part is kind of contained in the first. But I'd agree that being explicit is better. Maybe something like

```
This is a classifier that models the data in each class `k` as samples drawn from a multivariate Gaussian distribution with mean `\\mu_k` and covariance `\\Sigma_k`. This model therefore considers the data as real-valued. 
```

(note that the "considers as" is exactly what MLJ does when it takes the scitype to be continuous; you could have ordered data (1, 2, 1, 1) that is then coerced to continuous (1.0, 2.0, 1.0, 1.0)).

File: src/MLJNaiveBayesInterface.jl [hunk 2]
```diff
@@ -141,4 +141,125 @@ MMI.is_pure_julia(::Type{<:MultinomialNBClassifier}) = true
...
+The fields of `fitted_params(mach)` are:
+
+- `c_counts`: A dictionary containing the observed count of each input class.
+- `c_stats`: A dictionary containing observed statistics on each input class. Each class is
+  represented by a `DataStats` object, with the following fields:
+    - `n_vars`: The number of variables used to describe the class's behavior.
+    - `n_obs`: The number of times the class is observed.
+    - `obs_axis`: The axis along which the observations were computed.
+- `gaussians`: A per class dictionary of gaussians, each representing the distribution of
+  the class.
```
Comment (tlienart) [inline src/MLJNaiveBayesInterface.jl]: Might be good to add a note about how that object is represented (I'm assuming it's a Distribution.Normal or something like this)

File: src/MLJNaiveBayesInterface.jl [hunk 3]
```diff
@@ -141,4 +141,125 @@ MMI.is_pure_julia(::Type{<:MultinomialNBClassifier}) = true
...
+[`MultinomialNBClassifier`](@ref)
+"""
+GaussianNBClassifier
+"""
+$(MMI.doc_header(MultinomialNBClassifier))
+
+
+`MultinomialNBClassifier`: MultinomialNBClassifier Naive Bayes classifier.
+Assumes variables have a multinomial distribution. Good for counts and text classification.
+
```
Comment (tlienart) [inline src/MLJNaiveBayesInterface.jl:217]: Assuming you agree with my suggestion above, you could adapt it for this one too.

File: src/MLJNaiveBayesInterface.jl [hunk 4]
```diff
@@ -141,4 +141,125 @@ MMI.is_pure_julia(::Type{<:MultinomialNBClassifier}) = true
...
+The fields of `fitted_params(mach)` are:
+
+- `c_counts`: A dictionary containing the observed count of each input class.
+- `c_stats`: A dictionary containing observed statistics on each input class. Each class is
+  represented by a `DataStats` object, with the following fields:
+    - `n_vars`: The number of variables used to describe the class's behavior.
+    - `n_obs`: The number of times the class is observed.
+    - `obs_axis`: The axis along which the observations were computed.
+- `gaussians`: A per class dictionary of gaussians, each representing the distribution of
```
Comment (tlienart) [inline src/MLJNaiveBayesInterface.jl]: multinomials here I guess?

**Review by tlienart — COMMENTED** [review]
Cool thanks for this! I've added a few comments though a more serious review will likely come from the active maintainers.

File: src/MLJNaiveBayesInterface.jl (hunk 1 above)
Comment (josephsdavid) [reply src/MLJNaiveBayesInterface.jl:149]: > Cool thanks for this! I've added a few comments though a more serious review will likely come from the active maintainers.

Awesome! Thank you for your reviews 😄

**Comment by josephsdavid** [comment]
> Cool thanks for this! I've added a few comments though a more serious review will likely come from the active maintainers.

Thank you for your reviews! 😄

**Comment by ablaom** [comment]
@josephsdavid I've not yet reviewed, but here's a suggestion for the multinomial example (it needs #9):

```julia
using MLJ
import TextAnalysis

CountTransformer = @load CountTransformer pkg=MLJText
MultinomialNBClassifier = @load MultinomialNBClassifier pkg=NaiveBayes

tokenized_docs = TextAnalysis.tokenize.([
    "I am very mad. You never listen.",
    "You seem to be having trouble? Can I help you?",
    "Our boss is mad at me. I hope he dies.",
    "His boss wants to help me. She is nice.",
    "Thank you for your help. It is nice working with you.",
    "Never do that again! I am so mad. ",
])

sentiment = [
    "negative",
    "positive",
    "negative",
    "positive",
    "positive",
    "negative",
]

mach1 = machine(CountTransformer(), tokenized_docs) |> fit!

# matrix of counts:
X = transform(mach1, tokenized_docs)

# to ensure scitype(y) <: AbstractVector{<:OrderedFactor}:
y = coerce(sentiment, OrderedFactor)

classifier = MultinomialNBClassifier()
mach2 = machine(classifier, X, y)
fit!(mach2, rows=1:4)

# probabilistic predictions:
y_prob = predict(mach2, rows=5:6) # distributions
pdf.(y_prob, "positive") # probabilities for "positive"
log_loss(y_prob, y[5:6])

# point predictions:
yhat = mode.(y_prob) # or `predict_mode(mach2, rows=5:6)`

```

File: src/MLJNaiveBayesInterface.jl (hunk 2 above)
Comment (josephsdavid) [reply src/MLJNaiveBayesInterface.jl]: Should be resolved! Thank you!

File: src/MLJNaiveBayesInterface.jl (hunk 3 above)
Comment (josephsdavid) [reply src/MLJNaiveBayesInterface.jl:217]: Yep! Sounds good

File: src/MLJNaiveBayesInterface.jl (hunk 4 above)
Comment (josephsdavid) [reply src/MLJNaiveBayesInterface.jl]: This is actually a typo here on my part, should be resolved on latest commit!

**Comment by josephsdavid** [comment]
> @josephsdavid I've not yet reviewed, but here's a suggestion for the multinomial example (it needs #9):

 for now i just `MLJ.table`ed the data and it seems to work fine!

**Comment by ablaom** [comment]
Attention @ablaom

**Comment by ablaom** [comment]
Thanks @josephsdavid for your contribution. Great to have another out of the way.

**Pull request merged**